dockerfiles/models-gpu/Dockerfile

```diff
@@ -131,36 +128,18 @@ RUN conda install -n base -c pytorch pytorch torchvision cudatoolkit=10.1
 
 RUN ldconfig /usr/local/cuda-10.1/targets/x86_64-linux/lib/stubs && \
     HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_GPU_BROADCAST=NCCL HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITH_PYTORCH=1 \
-        python -m pip install --no-cache-dir horovod && \
```

Comment:
Why you removed the `--no-cache-dir` ?
now that I'm aware of this flag I would say we should ALWAYS pip install with it, it helps save space in the images, which is very important since our images are HUGE
so unless good reason not to, at least change all the pip install in all the docker files you touch in this PR to have this flag

Reply:
https://stackoverflow.com/questions/45594707/what-is-pips-no-cache-dir-good-for/45594808

I would like to have caching of layers so that they don't need to be re-downloaded over and over again.  It is part of a longer plan to cut the build time down dramatically.

Reply:
I believe there are circle-ci features that make this possible, although I only found out yesterday that they support buildkit layers only on their pay-for service

Reply:
@yjb-ds as far as I understand the `--no-cache-dir` as nothing to do with build time, only with the image size which will be nice to decrease. anyways let's leave it as is for now, and do optimizations in a dedicated PR